server/api/oauth_session.py

```diff
@@ -34,28 +46,24 @@ def __init__(
 
     async def refresh_tokens(self) -> None:
         assert self.refresh_token is not None
+        if not self.token_url.startswith('https://') and 'OAUTHLIB_INSECURE_TRANSPORT' not in os.environ:
+            raise InsecureTransportError()
         data = {
             'grant_type': 'refresh_token',
             'client_id': self.client_id,
             'client_secret': self.client_secret,
             'refresh_token': self.refresh_token
         }
-        creds = await self._make_request(
-            data=data,
-        )
+        creds = await self._make_request(data=data)
         self.update_tokens(creds)
 
     def update_tokens(self, creds: Dict[str, str]) -> None:
         self.token = creds['access_token']
-        refresh_token = creds.get('refresh_token')
-        if refresh_token:
-            self.refresh_token = refresh_token
-            expires_in = int(creds['expires_in'])
-            asyncio.create_task(self._schedule_refresh(expires_in - 5))
-
-    async def _schedule_refresh(self, wait_time):
-        await asyncio.sleep(wait_time)
-        await self.refresh_tokens()
+        self.refresh_token = creds.get('refresh_token')
+        expires_in = creds.get('expires_in')
+        if expires_in is not None:
```

Comment:
When will this be None? Can it happen that the API decides to send tokens without an expiry field and then you will consider all of them to be expired immediately?

Reply:
Well, it's data supplied by an external source so anything can happen. AFAIK it shouldn't happen though